wgpu-hal/src/dx12/command.rs

```diff
@@ -1192,3 +1193,13 @@ impl crate::CommandEncoder<super::Api> for super::CommandEncoder {
         };
     }
 }
+
+impl Drop for super::CommandEncoder {
+    fn drop(&mut self) {
+        // DX12 throws an exception when an encoder is dropped without being closed.
+        // To prevent this, we explicitiy call discard_encoding.
+        unsafe {
+            self.discard_encoding();
+        }
```

Comment:
🎤Registering @cwfitzgerald's [concern](https://github.com/gfx-rs/wgpu/pull/4023#issuecomment-1680915302) as a spanned conversation.

Reply:
Hmm, this `Drop` implementation appears to cause at least one compile regression in CI: https://github.com/gfx-rs/wgpu/actions/runs/5882808070/job/15954619239?pr=4023